### ISSUE TITLE
Bounded distribution does not work if bounds are not arrays

### ISSUE THREAD

**Issue by kiudee** [pymc-devs/pymc#1672]
Minimum example:
```python
import pymc3 as pm
with pm.Model() as mod:
    PositiveNormal = pm.Bound(pm.Normal, lower=0)
    a = PositiveNormal('a', mu=1, sd=2)
```
Gives the error:
```python
pymc3\distributions\continuous.py in __init__(self, distribution, lower, upper, transform, *args, **kwargs)
   1122 
   1123         if transform == 'infer':
-> 1124             self.transform, self.testval = self._infer(lower, upper)
   1125 
   1126     def _infer(self, lower, upper):

pymc3\distributions\continuous.py in _infer(self, lower, upper)
   1146         elif not self._isinf(_lower) and self._isinf(_upper):
   1147             transform = transforms.lowerbound(lower)
-> 1148             if (testval <= _lower).any():
   1149                 testval = lower + 1
   1150         elif self._isinf(_lower) and not self._isinf(_upper):

AttributeError: 'bool' object has no attribute 'any'
```

Because both `testval` and `_lower` are not arrays.

**Comment by ColCarroll** [comment]
Interesting -- I can confirm this breaks on master, but passes on 3.1.

**Comment by kiudee** [comment]
Yes, it appears it was fixed in 3.1:
```python
        if transform == 'infer':

            default = self.dist.default()

            if not np.isinf(lower) and not np.isinf(upper):
                self.transform = transforms.interval(lower, upper)
                if default <= lower or default >= upper:
                    self.testval = 0.5 * (upper + lower)

            if not np.isinf(lower) and np.isinf(upper):
                self.transform = transforms.lowerbound(lower)
                if default <= lower:
                    self.testval = lower + 1

            if np.isinf(lower) and not np.isinf(upper):
                self.transform = transforms.upperbound(upper)
                if default >= upper:
                    self.testval = upper - 1
```
